repo/ecc/ecc_rs_bw.go

```diff
@@ -0,0 +1,186 @@
+		shareSize = 1024
+		blocks = CeilInt(length, r.RequiredShares*shareSize)
+	} else {
+		blocks = 16
+		shareSize = CeilInt(length, r.RequiredShares*blocks)
+	}
+	originalSize = length
+	return
+}
+
+func (r *RsBwEcc) ComputeSizesFromStored(length int) (blocks, shareSize, originalSize int) {
+	if r.FixedShareSize {
+		shareSize = 1024
+		blocks = CeilInt(length, (r.RequiredShares+r.RedundantShares)*shareSize)
+	} else {
+		blocks = 16
+		shareSize = CeilInt(length, (r.RequiredShares+r.RedundantShares)*blocks)
+	}
+	originalSize = length - r.RedundantShares*shareSize*blocks
+	return
+}
+
+func (r *RsBwEcc) Encrypt(input gather.Bytes, contentID []byte, output *gather.WriteBuffer) error {
+	blocks, shareSize, originalSize := r.ComputeSizesFromOriginal(input.Length())
+	requiredSize := r.RequiredShares * shareSize
+
+	var inputBuffer gather.WriteBuffer
+	defer inputBuffer.Close()
+
+	inputBytes := inputBuffer.MakeContiguous(requiredSize * blocks)
+
+	copied := input.AppendToSlice(inputBytes[:0])
+
+	if len(copied) < len(inputBytes) {
+		clear(inputBytes[len(copied):])
+	}
+
+	inputPos := 0
+
+	for b := 0; b < blocks; b++ {
+		block := inputBytes[inputPos : inputPos+requiredSize]
+		inputPos += requiredSize
+
+		err := r.Fec.Encode(block, func(share infectious.Share) {
+			if share.Number >= r.RequiredShares {
+				output.Append(share.Data)
+			}
+		})
+		if err != nil {
+			return errors.Wrap(err, "Error computing ECC")
+		}
+	}
+
+	output.Append(inputBytes[:originalSize])
+
+	return nil
+}
+
+func (r *RsBwEcc) Decrypt(input gather.Bytes, contentID []byte, output *gather.WriteBuffer) error {
+	blocks, shareSize, originalSize := r.ComputeSizesFromStored(input.Length())
+	requiredSize := r.RequiredShares * shareSize
+	redundantSize := r.RedundantShares * shareSize
+
+	var inputBuffer gather.WriteBuffer
+	defer inputBuffer.Close()
+
+	inputBytes := inputBuffer.MakeContiguous((requiredSize + redundantSize) * blocks)
+
+	copied := input.AppendToSlice(inputBytes[:0])
+
+	if len(copied) < len(inputBytes) {
+		clear(inputBytes[len(copied):])
+	}
+
+	eccBytes := inputBytes[:redundantSize*blocks]
+	dataBytes := inputBytes[redundantSize*blocks:]
+
+	shares := make([]infectious.Share, r.RequiredShares+r.RedundantShares)
+	for i := 0; i < r.RequiredShares; i++ {
+		shares[i].Number = i
+	}
+	for i := 0; i < r.RedundantShares; i++ {
+		shares[r.RequiredShares+i].Number = r.RequiredShares + i
+	}
+
+	dataPos := 0
+	eccPos := 0
+	for b := 0; b < blocks; b++ {
+		for i := 0; i < r.RequiredShares; i++ {
+			shares[i].Data = dataBytes[dataPos : dataPos+shareSize]
+			dataPos += shareSize
+		}
+		for i := 0; i < r.RedundantShares; i++ {
+			shares[r.RequiredShares+i].Data = eccBytes[eccPos : eccPos+shareSize]
+			eccPos += shareSize
+		}
+
+		err := r.Fec.Correct(shares)
+		if err != nil {
+			return errors.Wrap(err, "Error computing ECC")
+		}
+	}
+
+	output.Append(dataBytes[:originalSize])
+
+	return nil
+}
+
+func (r *RsBwEcc) Overhead() int {
+	return 0
```

Comment:
this should panic() to ensure we don't call it. We need to refactor this and fix names anyway in separate PR.

Reply:
I had to implement it because `content.SharedManager.setupReadManagerCaches` calls it:
```
	sm.committedContents = newCommittedContentIndex(caching,
		uint32(sm.format.Encryptor().Overhead()),
		sm.format.WriteIndexVersion(),
		sm.enc.getEncryptedBlob,
		sm.namedLogger("committed-content-index"),
		caching.MinIndexSweepAge.DurationOrDefault(DefaultIndexCacheSweepAge))
```